synapse/storage/databases/main/events_worker.py

```diff
@@ -1936,34 +1948,19 @@ def is_event_next_to_gap_txn(txn: LoggingTransaction) -> bool:
                 /* Check to make sure the event referencing our event in question is not rejected */
                 LEFT JOIN rejections ON event_edges.event_id = rejections.event_id
                 WHERE
-                    event_edges.room_id = ?
-                    AND event_edges.prev_event_id = ?
+                    event_edges.prev_event_id = ?
```

Comment:
Again, this points to brokenness if someone rolls forward and back again. This is only used by the experimental [MSC3030](https://github.com/matrix-org/matrix-doc/pull/3030) support, so this is definitely an edge case.

Reply:
^ this comment is in reference to https://github.com/matrix-org/synapse/pull/12893#discussion_r883463184 when someone rolls forward and backward between Synapse versions.